src/mpi/coll/allgather/allgather_inter_local_gather_remote_bcast.c

```diff
@@ -25,7 +25,7 @@ int MPIR_Allgather_inter_local_gather_remote_bcast(const void *sendbuf, int send
 {
     int rank, local_size, remote_size, mpi_errno = MPI_SUCCESS, root;
     int mpi_errno_ret = MPI_SUCCESS;
-    MPI_Aint sendtype_sz;
+    MPI_Aint sendtype_sz = 0;
```

Comment:
future reviewer will wonder why this variable needs initialization but others don't. Do do not fix it here, rather fix it by add an `else` block in the later code and add a comment explaining the reasons -- that 1. we need it to silence compiler warnings, 2. the compiler was correct but non-root process ignores recv buffer in `gather` and ignores send but in `scatter`.

Reply:
Also, in `ialltoall/ialltoall_intra_brucks.c`, `recvtype_sz`, `newtype_sz` were declared as `int`, @jain-surabhi-23 could you add a commit change that into `MPI_Aint`?

Reply:
@hzhou I actually like the fix I originally have because it makes the compiler, klocwork/coverity happy, handles every case (as the variable is initialized in the beginning) and it is neater to look at instead of having several `else` blocks. So I won't be making any changes here. I understand if you want to do it your way, since you wrote the code originally :)

Reply:
I am OK if you prefer your code, but can we keep a discussion?

Say some one is reviewing this code and ask you the following questions, can you answer easily (assuming several months later)? --

* Why `sendtype_sz` is initialized to `0`,  while other variables are left uninitialized? Especially, why a very same or similar variable such as `recvtype_sz` or `recvtype_extent` in another file is still left uninitialized? What are the difference between them?

* Since there is no comment, the answer that "to quench (specific) compiler warning" is not that obvious. But let's say you have very good memory and remembers what is about, the next question is why the compiler warns? Is the warning correct (indicates a bug)?

* I find the above question not trivial. In fact, in this case, the compiler is absolutely correct identifying a case that a variable is being used without initialization, and we are setting a default to 0 for those case. Then the following question is: does the value 0 makes sense here? Of course the value 0 makes no sense here. This is where a future naive reviewer may get into a real confusion.

* Now of course in this case it actually is not a bug even though the compiler warns correctly and the default value `0` here is irrelevant and only purpose is to silence the compiler. The reason lies in the special semantics of `MPI_Gather` and `MPI_Scatter`.

Don't you think we should do something to avoid these future confusion and make the code a bit more explicit?